plugin/pkg/utils/miscellaneous_test.go

```diff
@@ -82,10 +84,64 @@ var _ = Describe("Miscellaneous", func() {
 
 	DescribeTable("#IsSeedUsedByShoot",
 		func(seedName string, expected bool) {
-			Expect(utils.IsSeedUsedByShoot(seedName, shoots)).To(Equal(expected))
+			Expect(IsSeedUsedByShoot(seedName, shoots)).To(Equal(expected))
 		},
 		Entry("is used by shoot", "seed1", true),
 		Entry("is used by shoot in migration", "seed2", true),
 		Entry("is unused", "seed3", false),
 	)
+
+	Describe("#ValidateZoneRemovalFromSeeds", func() {
+		var (
+			seedName = "foo"
+			kind     = "foo"
+
+			coreInformerFactory coreinformers.SharedInformerFactory
+			shootLister         corelisters.ShootLister
+
+			oldSeedSpec, newSeedSpec *core.SeedSpec
+			shoot                    *core.Shoot
+		)
+
+		BeforeEach(func() {
+			coreInformerFactory = coreinformers.NewSharedInformerFactory(nil, 0)
+			shootLister = coreInformerFactory.Core().InternalVersion().Shoots().Lister()
+
+			oldSeedSpec = &core.SeedSpec{
+				Provider: core.SeedProvider{
+					Zones: []string{"1", "2"},
+				},
+			}
+			newSeedSpec = oldSeedSpec.DeepCopy()
+
+			shoot = &core.Shoot{
+				Spec: core.ShootSpec{
+					SeedName: &seedName,
+				},
+			}
+		})
+
+		It("should do nothing because no zone was removed and no shoots exist", func() {
+			Expect(ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, seedName, shootLister, kind)).To(Succeed())
+		})
+
+		It("should do nothing because no zone was removed even though shoots exist", func() {
+			Expect(coreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(shoot)).To(Succeed())
+
+			Expect(ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, seedName, shootLister, kind)).To(Succeed())
+		})
+
+		It("should do nothing because zone was removed and no shoots exist", func() {
+			newSeedSpec.Provider.Zones = []string{"2"}
+
+			Expect(ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, seedName, shootLister, kind)).To(Succeed())
+		})
+
+		It("should return an error because zone was removed even though shoots exist", func() {
+			newSeedSpec.Provider.Zones = []string{"2"}
+			Expect(coreInformerFactory.Core().InternalVersion().Shoots().Informer().GetStore().Add(shoot)).To(Succeed())
+
+			Expect(ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, seedName, shootLister, kind)).To(MatchError(ContainSubstring("cannot remove zones")))
+		})
+	})
```

Comment:
```suggestion
		It("should do nothing because a new zone was added", func() {
			newSeedSpec.Provider.Zones = []string{"1", "2", "3"}

			Expect(ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, seedName, shootLister, kind)).To(Succeed())
		})
	})
```